cpp/ql/lib/semmle/code/cpp/controlflow/IRGuards.qll

```diff
@@ -897,7 +965,7 @@ private predicate unary_simple_comparison_eq(
   not test.isGLValue() and
   not simple_comparison_eq(test, _, _, _, _) and
   not simple_comparison_lt(test, _, _, _) and
-  not test = any(SwitchInstruction switch).getExpression() and
+  op = test.getExpressionOperand() and
```

Comment:
Although this is changed in a later commit. What is the reason behind this change?

Reply:
Oh, right. That was a bug I temporarily introducered and forgot to remove from the commit. The `not` shouldn't have been removed.

But, as you noted, this code is removed in the next commit so 🤷‍♂️